submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sJobSubmitter.java

```diff
@@ -101,98 +101,78 @@ public String getSubmitterType() {
   }
 
   @Override
-  public Job submitJob(JobSpec jobSpec)
-      throws InvalidSpecException {
+  public Job createJob(JobSpec jobSpec) throws InvalidSpecException {
     Job job = null;
-
-    boolean success = createJob(JobSpecParser.parseJob(jobSpec));
-    if (success) {
-      job = new Job();
-      job.setName(jobSpec.getName());
-    } else {
-      LOG.error("Failed to create job." + jobSpec.toString());
+    try {
+      MLJob mlJob = JobSpecParser.parseJob(jobSpec);
```

Comment:
I think, you should first determine whether the job name already exists, and return an accurate error message if it exists.